quinn-proto/src/congestion.rs

```diff
@@ -1,27 +1,65 @@
 //! Logic for controlling the rate at which data is sent
 
-use std::time::{Duration, Instant};
+use crate::connection::paths::RttEstimator;
+use std::fmt::Debug;
+use std::time::Instant;
 
+mod bbr;
+mod bbr_bw_estimation;
+mod bbr_min_max;
 mod cubic;
 mod new_reno;
 
+pub use bbr::{BBRConfig, BBR};
 pub use cubic::{Cubic, CubicConfig};
 pub use new_reno::{NewReno, NewRenoConfig};
 
+/// To help differentiate between packet sizes and packet numbers
+pub type PacketNumber = u64;
+
 /// Common interface for different congestion controllers
-pub trait Controller: Send {
+pub trait Controller: Send + Debug {
+    fn on_sent(&mut self, _now: Instant, _bytes: u64) {}
+
     /// Packet deliveries were confirmed
     ///
     /// `app_limited` indicates whether the connection was blocked on outgoing
     /// application data prior to receiving these acknowledgements.
-    fn on_ack(&mut self, now: Instant, sent: Instant, bytes: u64, app_limited: bool, rtt: Duration);
+    fn on_ack(
+        &mut self,
+        _now: Instant,
```

Comment:
It's a trait. No need to add `_` for unused parameters

Reply:
my bad, cargo fix was complaining so I just let it do its thing.

Reply:
it's still not happy about it. I think it's because of the empty default impl. I'll add an #[allow(unused_variables)].